### PR TITLE
returning error if the Storage operation for secret management fails

### LINKED ISSUE #7591: BlobStorageSecretsRepository Write failure should fail key creation request
We're seeing cases where a PUT to admin/host/functionkeys/testkey0 succeed and return 201 created, even though there was an exception thrown from the blob persistence logic [here](https://github.com/Azure/azure-functions-host/blob/91b7d1c780f2e040d32b02b38f6af5bab4799c96/src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsRepository.cs#L191). So somewhere upstream we must be swallowing this exception. If the persistence fails, the PUT request should fail. 

Additionally, the error message is rather poor in these cases. We only log "There was an error performing a write operation on the Blob Storage Secret Repository. Please ensure the 'AzureWebJobsStorage' connection string is valid.". We should provide more details on the actual underlying storage exception, rather than offering the unhelpful bit on checking the storage connection string.

See recent CRI #251344724

### PR DESCRIPTION
**Pull request by soninaren**
### Issue describing the changes in this PR

resolves #7591

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs [hunk 1]
```diff
@@ -330,9 +330,16 @@ public async Task<bool> DeleteSecretAsync(string secretName, string keyScope, Sc
...
             {
-                if (t.IsFaulted && t.Exception.InnerException is KeyVaultErrorException)
+                if (t.IsFaulted)
                 {
-                    result = OperationResult.Forbidden;
+                    _logger.LogError("Error adding or updating secrets", t.Exception.InnerException);
+
+                    result = OperationResult.Error;
+
+                    if (t.Exception.InnerException is KeyVaultErrorException)
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs]: @alrod - It looks like you added this KeyVaultErrorException=>403 conversion code when you initially added the keyvault provider [here](https://github.com/Azure/azure-functions-host/commit/2b8c2b851e2a415d70b40e7d47bc415a0a82475a#diff-5edb54e856f748486383109589f7c8980f5d5a68bb4e0a809ff20d6374c53452R318). Is there a reason why we need to retain this? I'm thinking that across the board, if SecretManager receives a Fault updating secrets, it should just return a 500.

Comment (soninaren) [reply src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs]: agreed, @alrod any concerns changing error code to 500?

Comment (alrod) [reply src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs]: it should be 500, please change this

**Review by liliankasem — DISMISSED** [review]

**Review by mathewc — APPROVED** [review]

**Review by liliankasem — APPROVED** [review]

**Pull request merged**